e2e/module_input_test.go

```diff
@@ -0,0 +1,345 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-terraform-sync/api"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+const (
+	// config snippets for test cases: module and module_input fields
+	servicesInput = `
+	module = "./test_modules/local_instances_file"
+	module_input "services"{
+		// insert names or regexp field below
+		%s
+		filter = "Service.Tags not contains \"filtered\""
+		cts_user_defined_meta {
+			meta_key = "meta_value"
+		}
+	}`
+	consulKVInput = `
+	module = "./test_modules/consul_kv_file"
+	module_input "consul-kv" {
+		path = "key"
+		recurse = %t
+	}`
+)
+
+// TestModuleInput_Basic_CatalogServicesCondition exercises:
+// 1. basics of module input
+// 2. simultaneously module input with catalog-services condition
+//
+// Module input basic testing exercises:
+// - each type of module_input
+// - module_input configuration field details
+// - multiple module_inputs
+func TestModuleInput_Basic_CatalogServicesCondition(t *testing.T) {
+	t.Parallel()
+
+	testcases := []struct {
+		name        string
+		tempDir     string
+		inputConfig string
+		validation  func(string, string)
```

Comment:
This pattern of setting the validation steps as part of the test case is really neat!